### PR TITLE
web: Display helpful errors when Ruffle fails to initialize

### PR DESCRIPTION
**Pull request by Toad06**
This PR is intended to help users of the selfhosted build, when Ruffle fails to initialize. To do so, new error messages are implemented.

Messages likely need to be improved. I also wonder if there's a better way to detect certain errors (eg. 404 and CORS). If this gets merged, we'll have to mention about the CORS issue on the wiki.

### CODE REVIEW

File: web/packages/core/src/ruffle-player.ts [hunk 1]
```diff
@@ -780,42 +811,125 @@ export class RufflePlayer extends HTMLElement {
...
-            issueParameters = encodeURIComponent(errorArray.join(""));
+            issueBody = encodeURIComponent(errorArray.join(""));
         }
-        issueLink += issueParameters;
+        issueLink += issueBody;
 
         // Clears out any existing content (ie play button or canvas) and replaces it with the error screen
+        let errorBody, errorFooter;
+        switch (errorIndex) {
+            case 100:
```
Comment (Herschel) [inline web/packages/core/src/ruffle-player.ts:829]: Could we use a TypeScript enum to represent these error codes:
https://www.typescriptlang.org/docs/handbook/enums.html

**Comment by Herschel** [comment]
It'd be helpful if we displayed an specific error for CORS errors when loading a cross-domain SWF, but that will require some changes to the Rust side and could be for a future PR.

**Comment by Toad06** [comment]
Thanks for the review. You're right, it's definitely better to use an enum here rather than arbitrary values. I've made the change, hope the naming is correct.

For SWFs, yeah I'd prefer to delay this to a future PR since that implies the Rust side. But I suppose that for these errors, we could reuse the template currently used for the AS3 warning.

**Review by Herschel — APPROVED** [review]

**Comment by Herschel** [comment]
Yes, I think re-using the AS3 warning might be fine. I guess it's a question of "What should we do when an SWF file is not found" as well? (The original Flash Player simply displayed a blank canvas).

**Pull request merged**